whisper/cli/src/config/logger.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use fern::colors::{Color, ColoredLevelConfig};
+use rlog::{LevelFilter};
+use std::io;
+
+fn setup_logger(verbosity: u32, logging_to_file: bool) -> Result<(), fern::InitError> {
+  let colors_line = ColoredLevelConfig::new()
+    .error(Color::Red)
+    .warn(Color::Yellow)
+    .info(Color::BrightBlack)
+    .debug(Color::BrightBlack)
+    .trace(Color::BrightBlack);
+
+  let colors_level = colors_line.clone()
+    .info(Color::Green);
+
+  let mut base_config = fern::Dispatch::new();
+
+  base_config = match verbosity {
+    0 => {
+      base_config
+      .level(LevelFilter::Error)
+      .level_for("pretty_colored", LevelFilter::Error)
+    }
+    1 => {
+      base_config
+      .level(LevelFilter::Warn)
+      .level_for("pretty_colored", LevelFilter::Warn)
+    },
+    2 => {
+      base_config
+      .level(LevelFilter::Info)
+      .level_for("pretty_colored", LevelFilter::Info)
+    },
+    3 => {
+      base_config
+      .level(LevelFilter::Debug)
+      .level_for("pretty_colored", LevelFilter::Debug)
+    },
+    _ => {
```

Comment:
Does this mean the default level is `trace`? Should be `info`.

Reply:
No, funnily enough I discovered that the default is set in the docopt CLI configuration option comments, i.e. `  [default: 0]` below means the default for the `--logging` value is `0` (i.e. Error)
```
	-l, --logging LEVEL            Specify the logging level. Must conform to the same format as RUST_LOG [default: 0]
```

Reply:
Actually, yes, if the user specifies a different value it's use trace